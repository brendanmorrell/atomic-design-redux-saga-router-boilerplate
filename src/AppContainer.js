import React from 'react'
import { connect } from 'react-redux'
// import PropTypes from 'prop-types'
import { ThemeProvider } from 'styled-components'
import { Switch, Route, Redirect } from 'react-router'

import theme from './theme'
import GlobalStyle from './theme/GlobalStyle'

import SessionTimeoutContainer from './auth/containers/SessionTimeoutContainer'
import StaticIFrameRouter from './external/organisms/StaticIframeRouter'

import Header from './header/containers/Header'
import SidebarContainer from './sidebar/containers/SidebarContainer';
import MobileMenuContainer from './mobilemenu/containers/MobileMenuContainer';
import AuthUpdatePassword from './auth/containers/AuthUpdatePassword';

const AppContainer = ({ loggedIn }) => {
  return (
    <SessionTimeoutContainer>
      <ThemeProvider theme={theme}>
        <>
          <GlobalStyle />
          
          <Header />
          <SidebarContainer />
          <MobileMenuContainer />
          
          <Switch>
            <Route path="/updatepassword" component={AuthUpdatePassword} />
            <StaticIFrameRouter />
            {/* 
            <Route path="/login" component={LoginPageContainer} />
            <Route exact path="/" render={() => <Redirect to="/home" />} />
            <ProtectedRouteContainer path="/home" component={HomePageContainer} /> 
            */}
          </Switch>
        </>
      </ThemeProvider>
    </SessionTimeoutContainer>
  );
};

const mstp = state => ({ loggedIn: state.auth.loggedIn });

export default connect(
  mstp,
  null,
)(AppContainer);
