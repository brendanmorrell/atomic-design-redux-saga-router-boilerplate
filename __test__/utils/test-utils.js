import React from 'react';
import { render } from 'react-testing-library';
import { ThemeProvider } from 'styled-components';
import theme from '../../src/theme';
import { Router } from 'react-router';
import { Provider } from 'react-redux';
import { storeFactory, history } from '../../src/store';

export const store = storeFactory({});
const customRender = (node, options) =>
  render(
    <Provider store={store}>
      <Router history={history}>
        <ThemeProvider theme={theme}>{node}</ThemeProvider>
      </Router>
    </Provider>,
    options,
  );

// re-export everything
export * from 'react-testing-library';

// override render method
export { customRender as render };
