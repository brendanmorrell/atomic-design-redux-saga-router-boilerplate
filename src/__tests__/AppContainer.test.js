import React from 'react';
import { render, fireEvent, debugDOM, flushPromises } from 'test-utils';
import AppContainer from '../AppContainer';

test('snapshot', () => {
  const { container } = render(<AppContainer />);
  expect(container.firstChild).toMatchSnapshot();
});
