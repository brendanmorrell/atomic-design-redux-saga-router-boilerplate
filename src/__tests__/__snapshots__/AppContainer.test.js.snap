// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="HeaderContainer__Container-sc-12f71jv-0 dCeWzT"
>
  <div
    class="SelectLocationContainer__Container-sc-1udcmzh-0 drzxgz"
  >
    <div
      class="SelectLocationContainer__Content-sc-1udcmzh-1 kNvJnx"
    >
      <div
        class="SelectLocationContainer__Wrapper-sc-1udcmzh-2 eHbgbq"
      >
        <div
          class="YourLocation__Mobile-qudshk-0 hWrbcP"
        >
          <img
            alt="Select Location"
            src="target-address.svg"
          />
        </div>
        <div
          class="YourLocation__Desktop-qudshk-1 bBLJeZ"
        >
          <div
            class="SelectLocationButton__Container-sc-15as5d8-0 cOlapo"
            type="location"
          >
            <div
              class="SelectLocationButton__Mobile-sc-15as5d8-5 dxuTPW"
            >
              <div
                class="SelectLocationButton__Label-sc-15as5d8-2 jAGIFS"
              />
            </div>
            <div
              class="SelectLocationButton__Desktop-sc-15as5d8-4 kZhMlH"
            >
              <div
                class="SelectLocationButton__Header-sc-15as5d8-1 gWyOfy"
              >
                Your locations
              </div>
              <div
                class="SelectLocationButton__Label-sc-15as5d8-2 jAGIFS"
              >
                No Locations
              </div>
            </div>
            <div
              class="SelectLocationButton__Arrow-sc-15as5d8-3 jDRwNi"
            />
          </div>
        </div>
        <div
          class="SelectLocationMenu__Menu-sc-1035de0-0 ejIPLP"
          type="location"
        >
          <div
            class="YourLocation__Container-qudshk-2 juvteI"
          >
            <img
              class="YourLocation__Arrow-qudshk-3 gAazDP"
              src="arrow-up-red.svg"
            />
            <img
              class="YourLocation__Close-qudshk-4 jkqsPg"
              src="close-blk.svg"
            />
            <h1>
              Personalize your experience
            </h1>
            <h2>
              For the best experience, please update your location.
            </h2>
            <div
              class="YourLocation__Input-qudshk-5 jXHEJI"
            >
              <input />
              <img
                class="YourLocation__Target-qudshk-6 Wgjfh"
                src="target-address.svg"
              />
            </div>
            <button
              class="Button__Container-awnt0e-0 bJqBrT"
              color="primary"
              disabled=""
              style="margin-top: 10px; width: 152px;"
            >
              Update
            </button>
          </div>
        </div>
        <div
          class="SelectLocationButton__Container-sc-15as5d8-0 gYitdm"
          type="state"
        >
          <div
            class="SelectLocationButton__Mobile-sc-15as5d8-5 dxuTPW"
          >
            <div
              class="SelectLocationButton__Label-sc-15as5d8-2 jAGIFS"
            >
              NV
            </div>
          </div>
          <div
            class="SelectLocationButton__Desktop-sc-15as5d8-4 kZhMlH"
          >
            <div
              class="SelectLocationButton__Header-sc-15as5d8-1 gWyOfy"
            >
              Selected State
            </div>
            <div
              class="SelectLocationButton__Label-sc-15as5d8-2 jAGIFS"
            >
              No Store
            </div>
          </div>
          <div
            class="SelectLocationButton__Arrow-sc-15as5d8-3 jDRwNi"
          />
        </div>
        <div
          class="SelectLocationMenu__Menu-sc-1035de0-0 gDkHuG"
          type="state"
        >
          <div
            class="SelectedState__State-sc-1cbtdyi-0 IJJwp"
          >
            Arizona
          </div>
          <div
            class="SelectedState__State-sc-1cbtdyi-0 IJJwp"
          >
            California
          </div>
          <div
            class="SelectedState__State-sc-1cbtdyi-0 IJJwp"
          >
            Florida
          </div>
          <div
            class="SelectedState__State-sc-1cbtdyi-0 IJJwp"
          >
            Illinois
          </div>
          <div
            class="SelectedState__State-sc-1cbtdyi-0 IJJwp"
          >
            Nevada
          </div>
          <div
            class="SelectedState__State-sc-1cbtdyi-0 IJJwp"
          >
            New York
          </div>
        </div>
        <div
          class="SelectLocationButton__Container-sc-15as5d8-0 dKdrXZ"
          type="store"
        >
          <div
            class="SelectLocationButton__Mobile-sc-15as5d8-5 dxuTPW"
          >
            <div
              class="SelectLocationButton__Header-sc-15as5d8-1 gWyOfy"
            >
              Store
            </div>
            <div
              class="SelectLocationButton__Label-sc-15as5d8-2 jAGIFS"
            >
              No Store
            </div>
          </div>
          <div
            class="SelectLocationButton__Desktop-sc-15as5d8-4 kZhMlH"
          >
            <div
              class="SelectLocationButton__Header-sc-15as5d8-1 gWyOfy"
            >
              Selected Store
            </div>
            <div
              class="SelectLocationButton__Label-sc-15as5d8-2 jAGIFS"
            >
              No Store
            </div>
          </div>
          <div
            class="SelectLocationButton__Arrow-sc-15as5d8-3 jDRwNi"
          />
        </div>
        <div
          class="SelectLocationMenu__Menu-sc-1035de0-0 eGhacg"
          type="store"
        >
          <div
            class="SelectedStore__Details-sc-114umlg-0 fphMIP"
          >
            See Selected Store Details
          </div>
          <div
            class="SelectedStore__Store-sc-114umlg-1 hWMEoo"
          >
            Monterey Bay - Seaside
          </div>
          <div
            class="SelectedStore__Store-sc-114umlg-1 hWMEoo"
          >
            San Jose - Central (N 10th & Commercial)
          </div>
          <button
            class="Button__Container-awnt0e-0 fFXPhR"
            color="primary"
          >
            View All Stores
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="SelectLocationContainer__Overlay-sc-1udcmzh-3 fQcvAN"
  />
  <div
    class="NavigationHeaderContainer__Container-sc-15wz8bx-0 bfEDXE"
  >
    <div
      class="NavigationHeaderContainer__Content-sc-15wz8bx-1 eKEtxa"
    >
      <img
        class="NavigationHeaderContainer__Logo-sc-15wz8bx-2 gRjkhR"
        data-testid="/"
        src="logo.png"
      />
      <div
        class="NavigationBar__Container-sc-1mqwq84-0 LeeJg"
      >
        <div
          class="NavigationLink__Container-zdydky-0 jIDNwV"
        >
          <div
            class="NavigationLink__Title-zdydky-1 jubtma"
          >
            <a
              data-testid="/shop"
              href="/shop"
            >
              Shop
            </a>
          </div>
        </div>
        <div
          class="NavigationLink__Container-zdydky-0 jIDNwV"
        >
          <div
            class="NavigationLink__Title-zdydky-1 jubtma"
          >
            <span>
              Our Brands
            </span>
          </div>
        </div>
        <div
          class="NavigationLink__Container-zdydky-0 jIDNwV"
        >
          <div
            class="NavigationLink__Title-zdydky-1 jubtma"
          >
            <a
              data-testid="/apparel"
              href="/apparel"
            >
              Apparel
            </a>
          </div>
        </div>
        <div
          class="NavigationLink__Container-zdydky-0 jIDNwV"
        >
          <div
            class="NavigationLink__Title-zdydky-1 jubtma"
          >
            <a
              data-testid="/ember"
              href="/ember"
            >
              EMBER
            </a>
          </div>
        </div>
        <div
          class="NavigationLink__Container-zdydky-0 jIDNwV"
        >
          <div
            class="NavigationLink__Title-zdydky-1 jubtma"
          >
            <a
              data-testid="/investors"
              href="/investors"
            >
              Investors
            </a>
          </div>
        </div>
      </div>
      <div
        class="NavigationBar__Overlay-sc-1mqwq84-1 jzhFpb"
      />
      <div
        class="NavigationHeaderContainer__Auth-sc-15wz8bx-3 hCCDtr"
      >
        Sign Up
      </div>
      <div
        class="NavigationHeaderContainer__Auth-sc-15wz8bx-3 hCCDtr"
        style="right: 120px;"
      >
        Profile
      </div>
      <div
        class="NavigationHeaderContainer__MobileMenu-sc-15wz8bx-4 kPVmxG"
      >
        <img
          alt="Menu"
          src="mobile-menu.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
